packages/check-bundle/test/test-check-bundle.js

```diff
@@ -0,0 +1,64 @@
+// @ts-check
+import '@endo/init/pre-bundle-source.js';
+import '@endo/init';
+import test from 'ava';
+import * as url from 'url';
+import * as crypto from 'crypto';
+import bundleSource from '@endo/bundle-source';
+import { checkBundle } from '../lite.js';
+
+const fixture = url.fileURLToPath(
+  new URL('fixture/main.js', import.meta.url).toString(),
+);
+
+/** @param {Uint8Array} bytes */
+const computeSha512 = bytes => {
+  const hash = crypto.createHash('sha512');
+  hash.update(bytes);
+  return hash.digest().toString('hex');
+};
+
+test('bundle and check get export package', async t => {
+  const bundle = await bundleSource(fixture, 'getExport');
+  await t.throwsAsync(checkBundle(bundle, computeSha512, 'fixture/main.js'), {
+    message: /checkBundle cannot determine hash of bundle with getExport moduleFormat because it is not necessarilly consistent/,
+  });
+});
+
+test('bundle and check nested evaluate package', async t => {
+  const bundle = await bundleSource(fixture, 'nestedEvaluate');
+  await t.throwsAsync(checkBundle(bundle, computeSha512, 'fixture/main.js'), {
+    message: /checkBundle cannot determine hash of bundle with nestedEvaluate moduleFormat because it is not necessarilly consistent/,
+  });
+});
+
+test('bundle and check endo zip base64 package', async t => {
+  const bundle = await bundleSource(fixture, 'endoZipBase64');
+  await checkBundle(bundle, computeSha512, 'fixture/main.js');
+  t.pass();
+});
+
+test('bundle and check corrupt endo zip base64 package', async t => {
```

Comment:
Hm, should we also test a package with the `endoZipBase64Sha512` property missing entirely, since that's similar to the way a lot of JWT libraries get broken (`alg: 'none'`)? Such bundles are fine for importing, but should not claim to be validatable.

Reply:
It's paranoia, but let's also test against a syntactically-valid but almost-certainly wrong hash, just 512/4=128 zeros, to get the test coverage deep enough into the implementation to be comparing hashes against hashes.